src/Exceptions/UnauthorizedException.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace Laravel\Horizon\Exceptions;
+
+use RuntimeException;
+
+class UnauthorizedException extends RuntimeException
```

Comment:
Should extends https://github.com/symfony/http-kernel/blob/6.3/Exception/UnauthorizedHttpException.php

You should be able to use renderable and typehint directly to Horizon exception to intercept the response.

Reply:
as @taylorotwell  mentioned, without it now the exception would render 500 error code.

Reply:
@crynobone that exception requires a challenge for the `WWW-Authenticate` header, which I'm assuming we don't want / need to set? I've extended the base `HttpException` and updated the status code to 401